solr/bin/solr

```diff
@@ -1697,7 +1702,7 @@ if [[ "$SCRIPT_CMD" == "auth" ]]; then
         fi
       done
   fi
-  run_tool auth "${AUTH_PARAMS[@]}" -solrUrl "$SOLR_URL_SCHEME://$SOLR_TOOL_HOST:${AUTH_PORT:-8983}/solr" -authConfDir "$SOLR_HOME" $VERBOSE
+  run_tool auth "${AUTH_PARAMS[@]}" --solr-url "$SOLR_URL_SCHEME://$SOLR_TOOL_HOST:${AUTH_PORT:-8983}/solr" --auth-conf-dir "$SOLR_HOME" $VERBOSE
```

Comment:
A general comment/question is that in main, most of the arg parsing is completely delegated to SolrCLI, while even in this PR we keep the `bin/solr` parsing and construction of args to pass to auth tool.

Question is whether we should do as in main. Well, main branch does not delegate `"auth"` tool parsing to java, but it does for many other tools. It would then look like
```bash
if [[ "$SCRIPT_CMD" == "auth" ]]; then
  run_tool auth "$@"
  exit $?
fi
```

Benefit would be that we don't need to do all the arg parsing changes to `bin/solr.cmd` too.

I did a test, but there seems to be quite some effort needed for this too. The AuthTool needs `--solr-include-file` to edit it, so bin/solr passes this arg today, but user is also supposed to be able to override this param. Thus for AuthTool to handle all parsing, we probably need to let bin/solr pass include-file path in a different param `--solr-include-file-default` so AuthTool could arbitrate between them?

So not sure what is most work - fix windows cmd script or make delegation work. The delegation path would be an effort that could be forward-ported to main...

Reply:
Really appreciate the thought that you are bringing to this effort.   I hope to see significantly more of the arg parsing logic in `bin/solr` and `bin/solr.cmd` move to Java code happen before 10x.  So maybe the delegation effort, while being more work, would help move that overall goal forward?